File: CommunityHub/CommunityHub.Domain/Entities/Event.cs
using CommunityHub.Domain.Enums;

namespace CommunityHub.Domain.Entities;

public class Event : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public DateTime EventDate { get; set; }
    public int MaxParticipants { get; set; }
    public string Venue { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public int Duration { get; set; }
    public string? ImagePath { get; set; }
    public EventStatus Status { get; set; }

    public Guid CommunityId { get; set; }
    public Community Community { get; set; } = null!;
    public ICollection<EventTag> Tags { get; set; } = [];
}

File: frontend/src/models/Event.ts
import { TagApiModel } from './Tag';

export interface Event {
  id: string;
  title: string;
  description: string;
  eventDate: string;
  maxParticipants: number;
  venue: string;
  address: string;
  status: string;
  duration: number;
  imagePath: string;
  latitude: number | null;
  longitude: number | null;
  communityName: string;
  communityId: string;
  tags: TagApiModel[];
}

export interface EventSearchRequest {
  keywords?: string;
  location?: string;
  category?: string;
  dateFrom?: string;
  dateTo?: string;
  page: number;
  pageSize: number;
}
